### DISCUSSION TITLE
Doubt about callbacks and eventloop

### DISCUSSION THREAD

**Discussion by juliancnn** [libuv/libuv#3907] [Q&A]
Hi guys, 

I'm new here, I'm researching and learning a bit about libuv and I have a question.
Suppose we have a single thread in the event loop, with a TCP handler (for example). When an event arrives it is processed by the callback. While it is processed and a second event arrives. If there is only one thread in the pool, the callback is executed after finishing the processing of the first event, right? That is what I understand from this graphic in the documentation:

![image](https://user-images.githubusercontent.com/1891958/218378337-0c491edf-2d55-4176-99c0-a8368f410626.png)
source: http://docs.libuv.org/en/v1.x/design.html

In the scenario where the same Handler receives events that may take considerable time to process and others that are processed faster, does libuv have a processing strategy?

Thank you very much for your help, have a great start of the week!

**Comment by bnoordhuis** [comment]
> the same Handler receives events that may take considerable time to process

I'll assume "considerable time" means "CPU time", not "blocks on I/O". Libuv doesn't really help you with that - it's a single-threaded event loop - but you can farm out long-running tasks to its thread pool with `uv_queue_work()`.
